sw/device/lib/arch/device.h

```diff
@@ -216,6 +216,11 @@ uint64_t to_cpu_cycles(uint64_t usec);
  */
 void device_fpga_version_print(void);
 
+/**
+ * Returns whether or not the current device is an FPGA.
+ */
+bool device_type_fpga(void);
```

Comment:
This is not a key that should be used for anything. Merely having an FPGA implementation does not logically imply specific properties. Let's not add this kind of shortcut when we'll need to directly describe capabilities for multi-top anyway.

Reply:
Do you mean we should stop checking "FPGA or DV-sim or Verilator" and directly check the properties causing the special-casing, e.g. clock speed, available pads, etc?

Reply:
> Do you mean we should stop checking "FPGA or DV-sim or Verilator" and directly check the properties causing the special-casing, e.g. clock speed, available pads, etc?

Yup!

> If so, I agree. I'm taking this shortcut for this experiment to save time, though.

No worries! I wanted to make sure, since we are carrying a lot of tech debt of this sort, and some of it is _just_ about coming due. xD

As we wrangle with that tech debt, we'll want to try to be more precise with the supported models and the mapping to a definition / query language.

Reply:
I've opened https://github.com/lowRISC/opentitan/issues/23558 to track this and removed `device_type_fpga` from this PR.